pkg/storegateway/bucket_index_reader.go

```diff
@@ -595,3 +596,51 @@ func (l *bucketIndexLoadedSeries) unsafeLoadSeriesForTime(ref storage.SeriesRef,
 	stats.seriesTouchedSizeSum += len(b)
 	return decodeSeriesForTime(b, lset, chks, skipChunks, mint, maxt)
 }
+
+type symbolsTable interface {
+	LookupLabelsSymbols([]symbolizedLabel) (labels.Labels, error)
+}
+
+type bucketIndexInflatedSeries struct {
```

Comment:
Why do we need a data structure at all? Isn't a standalone function like `inflateSeriesForTime()` enough (passing everything it needs)?

Reply:
this also keeps track of the buffered slices `[]symbolizedLabel` and `[]chunks.Meta`. I wanted to move this out of the other function